### PR TITLE
Fixing issues with empty help cards from #24998

### LINKED ISSUE #24998: Inline Help: Pressing Enter in the search field shows a broken / empty "Read More" screen
A gif is worth 💯 words (or something)
inline-help-enter-key-empty-result

#### Steps to reproduce
1. Starting at URL: https://wordpress.com (or anywhere on calypso)
2. Click the help icon to open inline help
3. Focus the search field
4. Hit enter (with or without text in the box)

#### What I expected
Either nothing (just the search submitted) or the topmost result to be expanded (probably the former).

Note: if you "select" a result with the Up or Down arrow key on the keyboard, pressing enter will expand the selected result as expected.

#### What happened instead
An "empty" excerpt / result screen was shown with a `Read More` button that does nothing when clicked.

#### Browser / OS version
Ffx / Chrome Mac High Sierra

#### Context / Source
Found while testing #24822

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
In this PR I am attempting to fix the issue reported by @jblz in #24998

The changes are small, but here is what I have changed in order to achieve the result, described in the "What I expected" section of the issue:

1. To actually fix the issue, I added a check in `client/blocks/inline-help/popover.jsx` in order to ensure that the secondary view (the one with the "Read More" button) is not being opened unless there is a selected result.
2. To slighly improve usability, every time new results are being displayed, the first one gets preselected. By doing so in `client/state/inline-help/reducer.js`, users can directly hit the Enter key after a successful search query, without having to use any arrows.

Since this is my first PR for the project and I do not find any tests for the block, I have not created any on my own. Still, I tested various scenarios and could not break anything.

As per the "Contributing to Calypso" guideline, I am pinging @lsinger to review the PR.

Fixes  #24998.

### CODE REVIEW

**Comment by lsinger** [comment]
@RadoslavGeorgiev thanks for the fix! Happy with (1) but regarding (2) as @jblz noted in the original issue ("probably the former"), I'd also prefer less magic than more. Let's *not* auto-select the first result — I think it's more confusing than helpful. 

Apart from that things look good!

**Comment by RadoslavGeorgiev** [comment]
@lsinger thanks for the reply :)

Done as requested:  `client/state/inline-help/reducer.js` has been reverted to the way it was - resetting back to `-1` upon request success. There are no changes in the first point.

File: client/blocks/inline-help/popover.jsx [hunk 1]
```diff
@@ -42,8 +42,12 @@ class InlineHelpPopover extends Component {
 	};
 
 	openResultView = event => {
+		const { selectedResult } = this.props;
 		event.preventDefault();
-		this.openSecondaryView( VIEW_RICH_RESULT );
+
+		if ( 'undefined' !== typeof selectedResult ) {
```
Comment (lsinger) [inline client/blocks/inline-help/popover.jsx]: Sorry for not including this earlier — I wonder whether we can use a lodash function here to make it a bit more terse, e.g. `isUndefined` or `isEmpty`. What do you think?

**Comment by lsinger** [comment]
Thanks a lot for your work @RadoslavGeorgiev, it's now merged and deployed! 🎉 😄

**Pull request merged**